challenge-099/dave-jacoby/perl/ch-1.pl

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+use utf8;
+use feature qw{say signatures };
+no warnings qw{experimental};
+
+my @arr;
+push @arr, [ 'abcde', 'a*e' ];
+push @arr, [ 'abcde', 'a*d' ];
+push @arr, [ 'abcde', '?b*d' ];
+push @arr, [ 'abcde', 'a?c*e' ];
+
+for my $n (@arr) {
+    my $p = pattern_match( $n->@* );
+    say $n->[0];
+    say $n->[1];
+    say $p;
+}
+
+sub pattern_match ( $S, $P ) {
+    my $pattern = $P;
+    $pattern =~ s/\*/.*/g;
+    $pattern =~ s/\?/.?/g;
```

Comment:
Hi, just happened to notice a bug here :)
```suggestion
    $pattern =~ s/\?/./g;
```
How about characters "." and "\" that are special in regexp but not in these patterns?

Reply:
Thanks for that.